src/main.rs

```diff
@@ -217,22 +226,22 @@ async fn handle_request(
         uri.path().to_owned()
     };
 
-    let path = match Path::try_from((method, trimmed_path.as_ref())).context(InvalidPath) {
+    let path = match Path::try_from((method, trimmed_path.as_ref())) {
         Ok(path) => path,
         Err(e) => {
             error!(
                 "Failed to parse path for {:?} {}: {:?}",
                 method, trimmed_path, e
             );
-            return Err(e);
+            return Err(RequestError::InvalidPath { source: e });
         }
     };
 
-    let bytes = match hyper::body::to_bytes(body).await.context(ChunkingRequest) {
+    let bytes = match hyper::body::to_bytes(body).await {
```

Comment:
This could be an if let

Reply:
It cannot due to the error being used, see https://rust-lang.github.io/rust-clippy/master/index.html#single_match_else as to why pedantic doesn't suggest this